lib/dictBuilder/cover.c

```diff
@@ -544,17 +544,17 @@ static int COVER_ctx_init(COVER_ctx_t *ctx, const void *samplesBuffer,
       totalSamplesSize >= (size_t)COVER_MAX_SAMPLES_SIZE) {
     DISPLAYLEVEL(1, "Total samples size is too large (%u MB), maximum size is %u MB\n",
                  (unsigned)(totalSamplesSize>>20), (COVER_MAX_SAMPLES_SIZE >> 20));
-    return 0;
+    return 72;
```

Comment:
You want to return an error here, like this

```
return ERROR(srcSize_wrong);
```

`ERROR(srcSize_wrong)` gets translated to `-(size_t)ZSTD_error_srcSize_wrong == -72`.

You can check for an error using

```
    size_t const ret = COVER_ctx_init(...);
    if (ZSTD_isError(ret)) {
        return ret;
    }
```

This "multiplexes" a return value and an error code into a single value. If the value is less than `ERROR(maxCode)`, which is a very large value (`(size_t)-120`), then the function succeeded and the returned value is a size. Otherwise the function failed and we can find the message with `ZSTD_getErrorName(ret)`.

You can take a look at `zstd_errors.h`, and `error_private.[hc]` for details on how it works.

Since we don't need a return value we can just return `0` on success.